splash/browser_tab.py

```diff
@@ -759,12 +825,72 @@ def on_initial_layout_completed(self):
     def on_url_changed(self, url):
         self.log("mainFrame().urlChanged %s" % qurl2ascii(url))
 
-    def log(self, message, min_level=None):
+    def log(self, message, min_level=None, encode=True):
         if min_level is not None and self.verbosity < min_level:
             return
 
-        if isinstance(message, unicode):
+        if encode and isinstance(message, unicode):
             message = message.encode('unicode-escape').decode('ascii')
```

Comment:
Log messages are encoded in order to make then ascii-safe - why is thi option needed? I think we should either encode all messages or encode none.